src/variants/lambdatreasury/storage.mligo

```diff
@@ -0,0 +1,185 @@
+// SPDX-FileCopyrightText: 2022 Tezos Commons
+// SPDX-License-Identifier: LicenseRef-MIT-TC
+
+#if !VARIANT_STORAGE
+#define VARIANT_STORAGE
+
+#include "variants/lambda/types.mligo"
+#include "error_codes.mligo"
+
+// Xtz transfer amount cannot be 0
+let zero_mutez_err_msg = "ZERO_MUTEZ"
+
+// Xtz transfer amount cannot be smaller than `min_xtz_amount`
+let too_small_xtz_err_msg = "LOW_XTZ"
+
+// Xtz transfer amount cannot be bigger than `max_xtz_amount`
+let too_large_xtz_err_msg = "HIGH_XTZ"
+
+let fail_proposal_check_ (msg : string) : unit =
+  ([%Michelson ({| { FAILWITH } |} : (nat * string) -> unit)]
+    (fail_proposal_check, msg) : unit)
+
+type config_proposal =
+  { frozen_scale_value : nat option
+  ; frozen_extra_value : nat option
+  ; slash_scale_value : nat option
+  ; slash_division_value : nat option
+  ; max_proposal_size : nat option
+  }
+
+type xtz_transfer =
+  [@layout:comb]
+  { amount : tez
+  ; recipient : address
+  }
+
+type token_id = nat
+// FA2 transfer types
+type transfer_destination =
+  [@layout:comb]
+  { to_ : address
+  ; token_id : token_id
+  ; amount : nat
+  }
+type transfer_item =
+  [@layout:comb]
+  { from_ : address
+  ; txs : transfer_destination list
+  }
+
+type token_transfer =
+  [@layout:comb]
+  { contract_address : address
+  ; transfer_list : transfer_item list
+  }
+
+type transfer_type =
+  [@layout:comb]
+  | Xtz_transfer_type of xtz_transfer
+  | Token_transfer_type of token_transfer
```

Comment:
These types are also the same between the two lambda variants, can they be moved to avoid duplication?

Reply:
Same as above, unable to import `common.mligo` module here because of circular imports.